### PR TITLE
add laszip/3.4.3

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **laszip/3.4.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/laszip/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
```
Comment (uilianries) [inline recipes/laszip/all/test_package/CMakeLists.txt]:
```suggestion
set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

https://github.com/LASzip/LASzip/blob/3.4.3/CMakeLists.txt#L47

File: recipes/laszip/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.options.shared:
+            self.cpp_info.defines.append("LASZIP_DYN_LINK")
```
Comment (uilianries) [inline recipes/laszip/all/conanfile.py]:
```suggestion
            self.cpp_info.defines.append("LASZIP_DYN_LINK")
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

ldd lib/liblaszip.so
	linux-vdso.so.1 (0x00007ffe6c4b3000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f7334bf7000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f7334ab1000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f7334a97000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f73348d1000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f7334e97000)

File: recipes/laszip/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (SpaceIm) [reply recipes/laszip/all/test_package/CMakeLists.txt]: unordered_map is not in public headers. It's a C API.

Comment (uilianries) [reply recipes/laszip/all/test_package/CMakeLists.txt]: fair enough

**Pull request merged**